File: CourseProjects/CourseProjects/VisualStudioProjects/Languages/Basic CS  Programming/Shapes/Square.cs
using System;

namespace Interfaces
{
	public class Square : Shape {
		public override void paint(int color) {}
		public override void draw(int size) {
			drawTopOrBottom(size);
			drawSides(size);
			drawTopOrBottom(size);
		}
		private void drawTopOrBottom(int size) {
			for(int i=0;i<size;i++) {
				Console.Write("+");
			}
			Console.WriteLine();
		}
		private void drawSides(int size) {
			for(int i=0;i<size;i++) {
				Console.Write("+");
				for(int x=2;x<size;x++) {
					Console.Write(" ");
				}
				Console.WriteLine("+");
			}
		}
	}
}

File: CourseProjects/CourseProjects/VisualStudioProjects/Languages/Basic VB Programming/Interfaces/Square.vb
Public Class Square
    Inherits Shape

    Public Overrides Sub paint(ByVal color As Integer)
    End Sub
    Public Overrides Sub draw(ByVal size As Integer)
        drawTopOrBottom(size)
        drawSides(size)
        drawTopOrBottom(size)
    End Sub
    Private Sub drawTopOrBottom(ByVal size As Integer)
        For i As Integer = 0 To size
            Console.Write("+")
        Next i
        Console.WriteLine()
    End Sub
    Private Sub drawSides(ByVal size As Integer)
        For i As Integer = 0 To size
            Console.Write("+")
            For x As Integer = 2 To size
                Console.Write(" ")
            Next x
            Console.WriteLine("+")
        Next i
    End Sub
End Class
